lib/Conversion/TorchToLinalg/Uncategorized.cpp

```diff
@@ -803,6 +803,23 @@ static Value createLinalgPayloadCalculationForElementwiseOp(
     Value other = convertScalarToDtype(b, loc, operands[1], dtype);
     return b.create<arith::DivFOp>(loc, self, other);
   }
+  if (auto remScalar = dyn_cast<AtenRemainderScalarOp>(op)) {
+    Type newResultType = converter->convertType(remScalar.getType())
+                             .cast<RankedTensorType>()
+                             .getElementType();
+
+    Value self = convertScalarToDtype(b, loc, payloadArgs[0], newResultType);
+    Value other = convertScalarToDtype(b, loc, operands[1], newResultType);
+    Value result;
+
+    if (newResultType.isa<mlir::FloatType>()) {
+      result = b.create<arith::RemFOp>(loc, self, other);
+    } else {
+      result = b.create<arith::RemSIOp>(loc, self, other);
+    }
```

Comment:
The `newResultType` is not guaranteed to be float or int. For example, it could be a `bool`. So, each branch of the `if` needs to check that the type is the right type, and the `else` should be used to `emitError` when the type is not one of the supported types

Reply:
yeah that makes sense.